### DISCUSSION TITLE
Question

### DISCUSSION THREAD

**Discussion by AlexFundorin** [taocpp/json#22] [Q&A] [answered]
How do I specify the path, where filename.json will be searched for?
Let's say, my c++ project is a DLL, which is in "c:\Plugin" folder. How do I properly access "c:\Plugin\JSON\filename.json" file?

P.S. I'm also wondering, why this warning appears:
```../internal/grammar.hpp(140): warning C4003: not enough actual parameters for macro 'max'```

Besides that warning, I had a couple of errors about std::min in the library. After changing them to min (I'm using namespace std", those errors are gone.

**Comment by ColinH** [comment]
The library does not manage any search paths, it would be application-specific which search paths to use, from which environment variable to take search paths etc. Usually all file access is relative to the current working directory of the application, i.e. the directory that the application was launched from.

Regarding the warning about `max` I'm not quite sure what's going on, it doesn't seem to occur anywhere. Which version of the library are you using? And what were the errors regarding `std::min()`? We do not test with `using namespace std` because we don't use `using` like that.

**Comment by ColinH** [comment]
The issues with `min` and `max` are probably because some Windows headers define them as macro. Fortunately there seems to be [a simple solution](https://stackoverflow.com/questions/5004858/stdmin-gives-error), does this work for you?

**Comment by AlexFundorin** [comment]
Thanks for the minmax solution. I'll try it later.

Is this ```value v = parse_file("c:\filename.json");``` a valid code, then?
Or ```value v = parse_file("../filename.json");```  this?
and also ```value v = parse_file("maps\filename.json");``` this?

**Comment by d-frey** [comment, marked as answer]
You need to be careful with escape sequences. If you want the filename to be the string `c:\filename.json`, you need to escape the `\`, your source code should thus contain `"c:\\filename.json"` (note: a double backslash).

**Comment by d-frey** [comment]
I've also just committed a change to protect against min/max macros, so the master branch should work even *without* `NOMINMAX`.

**Comment by AlexFundorin** [comment]
Thank you!

**Comment by d-frey** [comment]
You're welcome. Please test if your problems are solved (we are not Windows-users ourselves), feel free to re-open this issue if the fix is not working for you.

**Comment by AlexFundorin** [comment]
I've just replaced the files and it worked without any warnings or errors.
